src/main/java/com/minecolonies/core/colony/jobs/AbstractJobGuard.java

```diff
@@ -74,11 +77,29 @@ public void initEntityValues(AbstractEntityCitizen citizen)
         if (workBuilding instanceof AbstractBuildingGuards)
         {
             AttributeModifierUtils.addHealthModifier(citizen,
-              new AttributeModifier(GUARD_HEALTH_MOD_BUILDING_NAME, ((AbstractBuildingGuards) workBuilding).getBonusHealth(), AttributeModifier.Operation.ADDITION));
+                new AttributeModifier(GUARD_HEALTH_MOD_BUILDING_NAME, ((AbstractBuildingGuards) workBuilding).getBonusHealth(), AttributeModifier.Operation.ADDITION));
             AttributeModifierUtils.addHealthModifier(citizen,
-              new AttributeModifier(GUARD_HEALTH_MOD_CONFIG_NAME,
-                MineColonies.getConfig().getServer().guardHealthMult.get() - 1.0,
-                AttributeModifier.Operation.MULTIPLY_TOTAL));
+                new AttributeModifier(GUARD_HEALTH_MOD_CONFIG_NAME,
+                    MineColonies.getConfig().getServer().guardHealthMult.get() - 1.0,
+                    AttributeModifier.Operation.MULTIPLY_TOTAL));
+        }
+    }
+
+    @Override
+    public void onRemoval()
+    {
+        super.onRemoval();
+
+        if (this.getCitizen().getEntity().isPresent())
+        {
+            final Mob citizenEntity = getCitizen().getEntity().get();
+            // Unequip armor and weapons
+            citizenEntity.setItemSlot(EquipmentSlot.MAINHAND, ItemStackUtils.EMPTY);
```

Comment:
This shouldn't work =D If we wanna unequip, then we move the stuff from the armor slots in the citizendata to the normal inventory